src/shared/components/query/QueryStoreUtils.ts

```diff
@@ -25,14 +25,6 @@ export function queryParams(nonMolecularProfileParams:NonMolecularProfileQueryPa
                             molecularProfileParams:MolecularProfileQueryParams) {
     let params:CancerStudyQueryUrlParams = Object.assign({}, nonMolecularProfileParams, molecularProfileParams);
 
-    // Remove params with no value, because they may cause problems.
-    // For example, the server will always transpose if transpose_matrix is present, no matter the value.
-    for (let key in params) {
```

Comment:
doing this here meant that the routing store updateURL call would not get a value for this param, which would then allow any OLD param to be retained